diagnostic_tool/diagnostic_script.py

```diff
@@ -0,0 +1,165 @@
+import sys
+import blosc
+import dill
+import numpy as np
+from time import sleep, time
+from functools import partial
+from interbotix_xs_modules.xs_robot.arm import InterbotixManipulatorXS
+
+
+if sys.argv[1] == "deploy":
+    DEPLOY_MODE = True
+else:
+    DEPLOY_MODE = False
+
+
+class WX250sManipulator:
+    def __init__(self):
+        self.bot = InterbotixManipulatorXS("wx250s", "arm", "gripper")
+        self.arm = self.bot.arm
+        self.joint_names = self.arm.group_info.joint_names
+        self.num_joints = len(self.joint_names)
+
+    def get_gripper_tf(self):
+        return self.arm.T_sb
+
+    def get_joint_positions(self):
+        return self.arm.core.joint_states.position[:6]
+
+    # def set_joint_positions(self, joint_positions, moving_time):
+    #     pass
+
+    def create_right_angle(self, moving_time=3.0):
+        self.arm.set_joint_positions([0]*self.num_joints, moving_time=moving_time)
+
+    def point_straight_cartesian(self, moving_time=3.0):
+        self.create_right_angle(moving_time)
+
+        self.arm.set_ee_cartesian_trajectory(x=-0.1, z=-0.2, moving_time=moving_time)
+        print(self.get_joint_positions())
+
+    def point_straight(self, moving_time=3.0):
+        self.create_right_angle(moving_time)
+        if DEPLOY_MODE: sleep(0.5)
+
+        lower_jp = [0.0, -0.00020950394840690834, 0.9273922091469897, 0.0, -0.9271827051985841, 0.0]
+        self.arm.set_joint_positions(lower_jp, moving_time=moving_time)
+        if DEPLOY_MODE: sleep(0.5)
+
+    def compute_joint_trajectories(self, way_points, moving_time=4.0):
+        joint_trajectories = []
+
+        for i, (dx, dy, dz) in enumerate(way_points):
+            print("Computing joint positions for path {}".format(i+1))
+            jt = self.arm.set_ee_cartesian_trajectory(x=dx, y=dy, z=dz,
```

Comment:
What does `jt` mean here? In general, if the "original word" is not too long, it is recommended to just use the original word as (variable/function) names. The reason behind this is that the reader may have different habits/background, and this can potentially make the code more readable to everyone :)

For example, from line 38 I thought `set_ee_cartesian_trajectory` does not return anything. Here I am now assuming it returns something, but I cannot tell right away what it returns.

Line 56 and 57 suggest that `jt` is very likely to be a list of joint positions. Are the joint positions the resulting joint positions after `set_ee_cartesian_trajectory` finishes?

You could also add type hints to make it more readable:

```python
joint_trajectory: List[np.ndarray] = self.arm.set_ee_ ....
```

Reply:
This is a dev-file and is no longer exists.